CrossPlatform/BasePage.cs

```diff
@@ -14,7 +14,9 @@ public abstract class BasePage
         protected bool OnAndroid { get; private set; }
         protected bool OniOS { get; private set; }
 
-        protected Func<AppQuery, AppQuery> Trait { get; set; }
+        protected abstract Trait Trait { get; }
+
+        protected Func<AppQuery, AppQuery> CurrentTrait { get; private set; }
```

Comment:
I was on the fence about it. Our options are to keep it as is, or change the instances of `CurrentTrait` to `OnAndroid ? Trait.AndroidTrait : Trait.iOSTrait`.
I'm ok with it either way.